ietf/doc/views_statement.py

```diff
@@ -17,4 +29,247 @@ def serve_pdf(self, name, rev=None):
+        ("upload", "Upload content from file"),
+    ]
+    statement_submission = forms.ChoiceField(choices=ACTIONS, widget=forms.RadioSelect)
+    statement_file = forms.FileField(
+        label="Markdown or PDF source file to upload", required=False
+    )
+    statement_content = forms.CharField(
+        widget=forms.Textarea(attrs={"rows": 30}), required=False, strip=False
+    )
+
+    def clean(self):
+        def require_field(f):
+            if not self.cleaned_data.get(f):
+                self.add_error(f, forms.ValidationError("You must fill in this field."))
+                return False
+            else:
+                return True
+
+        submission_method = self.cleaned_data.get("statement_submission")
+        markdown_content = ""
+        if submission_method == "enter":
+            if require_field("statement_content"):
+                markdown_content = self.cleaned_data["statement_content"].replace(
+                    "\r", ""
+                )
+                default_content = render_to_string(
+                    "doc/statement/statement_template.md", {"settings": settings}
+                )
+                if markdown_content == default_content:
+                    raise forms.ValidationError(
+                        "The example content may not be saved. Edit it to contain the next revision statement content."
+                    )
+                if (
+                    markdown_content
+                    == "The current revision of this statement is in pdf format"
+                ):
+                    raise forms.ValidationError(
+                        "Not proceeding with the text noting that the current version is pdf. Did you mean to upload a new PDF?"
+                    )
+        elif submission_method == "upload":
+            if require_field("statement_file"):
+                content_type = self.cleaned_data["statement_file"].content_type
+                acceptable_types = (
+                    "application/pdf",
+                ) + settings.DOC_TEXT_FILE_VALID_UPLOAD_MIME_TYPES
+                if not content_type.startswith(
+                    acceptable_types
+                ):  # dances around decoration of types with encoding etc.
+                    self.add_error(
+                        "statement_file",
+                        forms.ValidationError(
+                            f"Unexpected content type: Expected one of {', '.join(acceptable_types)}"
+                        ),
+                    )
+                elif content_type != "application/pdf":
+                    markdown_content = get_cleaned_text_file_content(
+                        self.cleaned_data["statement_file"]
+                    )
+        if markdown_content != "":
+            try:
+                _ = markdown.markdown(markdown_content)
+            except Exception as e:
+                raise forms.ValidationError(f"Markdown processing failed: {e}")
```

Comment:
Might be better to log the exception instead of (or in addition to) exposing it to the UI, particularly if you're catching `Exception` instead of something more specific.

Reply:
well, what Exception might get thrown in this case that isn't "Markdown processing failed"?
It's exposed in the UI so the submitter can edit whatever they're sending in that markdown is blowing up on.